interfaces/builtin/unity8_pim_common.go

```diff
@@ -0,0 +1,221 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/release"
+)
+
+var unity8PimCommonPermanentSlotAppArmor = []byte(`
+# Description: Allow operating as the EDS service. Reserved because this
+#  gives privileged access to the system.
+
+# DBus accesses
+#include <abstractions/dbus-session-strict>
+
+dbus (send)
+	bus=session
+	path=/org/freedesktop/DBus
+	interface=org.freedesktop.DBus
+	member={Request,Release}Name
+	peer=(name=org.freedesktop.DBus, label=unconfined),
+
+dbus (send)
+	bus=session
+	path=/org/freedesktop/*
+	interface=org.freedesktop.DBus.Properties
+	peer=(label=unconfined),
+
+# Allow services to communicate with each other
+dbus (receive, send)
+	peer=(label="snap.@{SNAP_NAME}.*"),
+
+# Allow binding the service to the requested connection name
+dbus (bind)
+	bus=session
+	name="org.gnome.evolution.dataserver.Sources5",
+`)
+
+var unity8PimCommonConnectedSlotAppArmor = []byte(`
+# Allow service to interact with connected clients
+
+########################
+# SourceManager
+########################
+dbus (receive, send)
+	bus=session
+	path=/org/gnome/evolution/dataserver/SourceManager{,/**}
+	peer=(label=###PLUG_SECURITY_TAGS###),
+`)
+
+var unity8PimCommonConnectedPlugAppArmor = []byte(`
+# DBus accesses
+#include <abstractions/dbus-session-strict>
+
+dbus (send)
+	bus=session
+	path=/org/freedesktop/DBus
+	interface=org.freedesktop.DBus
+	member={Request,Release}Name
+	peer=(label=###SLOT_SECURITY_TAGS###),
```

Comment:
I didn't notice this one before. Why do connecting client need to RequestName and ReleaseName? I expected this in CommonPermanentSlotAppArmor, not for plugging clients.